x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/group_stats_aggregations.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { NamedAggregation } from '@kbn/grouping';
+import { DEFAULT_GROUP_STATS_AGGREGATION } from '../../alerts_table/alerts_grouping';
+import {
+  RULE_COUNT_AGGREGATION,
+  SEVERITY_SUB_AGGREGATION,
+} from '../../alerts_table/grouping_settings';
+
+const RULE_SIGNAL_ID_SUB_AGGREGATION = {
+  signalRuleIdSubAggregation: {
+    terms: {
+      field: 'signal.rule.id',
+    },
+  },
+};
+
+/**
+ * Returns aggregations to be used to calculate the statistics to be used in the`extraAction` property of the EuiAccordion component.
+ * It handles custom renders for the following fields:
+ * - signal.rule.id
+ * - kibana.alert.severity
+ * - kibana.alert.rule.name
+ * And returns a default set of aggregation for all the other fields.
+ *
+ * These go hand in hand with groupingOptions and groupPanelRenderers.
+ */
+export const groupStatsAggregations = (field: string): NamedAggregation[] => {
+  const aggMetrics: NamedAggregation[] = DEFAULT_GROUP_STATS_AGGREGATION('');
+
+  switch (field) {
+    case 'signal.rule.id':
+      aggMetrics.push(...[SEVERITY_SUB_AGGREGATION, RULE_COUNT_AGGREGATION]);
```

Comment:
minor nit, but why not just `aggMetrics.push(SEVERITY_SUB_AGGREGATION, RULE_COUNT_AGGREGATION);`

Reply:
No reason other than I copied the code from the alerts_table grouping code and didn't think about it 😆.
I can change it here and there as well for consistency!

Reply:
fixed in https://github.com/elastic/kibana/pull/216744/commits/c4cb1b7e72deba56f9fa335ee27825f0d17ad899